### PR TITLE
Fixed a storage key issue #129

### PR DESCRIPTION
**Pull request by dgofman**
Another fix on line 594. Defined a local variable srp_username, otherwise you overriding user value.

### CODE REVIEW

**Comment by furaiev** [comment]
Thank you, I’ll review on Sunday-Monday

**Comment by furaiev** [comment]
@dgofman thank you,
changes are available as of version 0.1.25+3

**Comment by dgofman** [comment]
@furaiev Thanks!

File: lib/src/cognito_user.dart [hunk 1]
```diff
@@ -246,12 +246,11 @@ class CognitoUser {
     final authParameters = {
       'REFRESH_TOKEN': refreshToken.getToken(),
     };
-    final keyPrefix = 'CognitoIdentityServiceProvider.${pool.getClientId()}';
     final lastUserKey = '$keyPrefix.LastAuthUser';
```
Comment (malkovro) [inline lib/src/cognito_user.dart:249]: You removed 
```dart
final keyPrefix = 'CognitoIdentityServiceProvider.${pool.getClientId()}';
```
And replaced it by 
```dart
  String get keyPrefix => 'CognitoIdentityServiceProvider.${pool.getClientId()}.$username';
```

Meaning that lastUserKey went from 
```dart
 'CognitoIdentityServiceProvider.${pool.getClientId()}.LastAuthUser';
```
To 
```dart
'CognitoIdentityServiceProvider.${pool.getClientId()}.$username.LastAuthUser';
```
Unfortunately, it's still referenced in [CognitoUserPool](https://github.com/furaiev/amazon-cognito-identity-dart-2/blob/4a0477dd236297100320154202b2fe8e255f2c19/lib/src/cognito_user_pool.dart#L81) as the former

**Review by malkovro — COMMENTED** [review]
I believe the `LastAuthUser` key should have been _protected_ from this refactor.

File: lib/src/cognito_user.dart (hunk 1 above)
Comment (dgofman) [reply lib/src/cognito_user.dart:249]: @malkovro That just a key name for the storage. I don't see any problems.
If you set your value using the key 'CognitoIdentityServiceProvider.PoolId1.Usernam1.**LastAuthUser**';
What's wrong to use the same key gettin your value back?

Comment (malkovro) [reply lib/src/cognito_user.dart:249]: How is the CognitoUserPool supposed to retreive the `LastAuthUser` username on line 81 with this key change ?

**Comment by dgofman** [comment]
There are two issues. One is mine I didn't realize amazon-cognito-identity-dart-2/lib/src/cognito_user_pool.dart hardcoded and using the same key. In the second one we are initializing storage in two contractors:

https://github.com/furaiev/amazon-cognito-identity-dart-2/blob/master/lib/src/cognito_user.dart#L53
https://github.com/furaiev/amazon-cognito-identity-dart-2/blob/master/lib/src/cognito_user_pool.dart#L44

In case you forgot to pass the same storage argument in one of them your 'lastUserKey' value will be always NULL.
For this reason, I did not catch this problem before. (I am using personal storage)
Now I am also dealing with this problem.

https://github.com/furaiev/amazon-cognito-identity-dart-2/pull/140

**Pull request merged**